certifier_service/simpleserver.go

```diff
@@ -376,93 +375,92 @@ func serviceThread(conn net.Conn, client string) {
-        }
-
-        // Service client connections.
-        for {
-                fmt.Printf("server: at accept\n")
-                conn, err = sock.Accept()
-                if err != nil {
-                        fmt.Printf("server: can't accept connection: %s\n", err.Error())
-                        continue
-                }
-                // Todo: maybe get client name and client IP for logging.
-                var clientName string = "blah"
-                go serviceThread(conn, clientName)
-        }
+	if !initCertifierService() {
+		fmt.Printf("server: failed to initialize server\n")
+		os.Exit(1)
+	}
+
+	var sock net.Listener
+	var err error
+	var conn net.Conn
+
+	// Listen for clients.
+	fmt.Printf("server: listening\n")
+	sock, err = net.Listen("tcp", serverAddr)
+	if err != nil {
+		fmt.Printf("server, listen error: ", err, "\n")
+		return
+	}
+
+	// Service client connections.
+	for {
+		fmt.Printf("server: at accept\n")
+		conn, err = sock.Accept()
+		if err != nil {
+			fmt.Printf("server: can't accept connection: %s\n", err.Error())
+			continue
+		}
+		// Todo: maybe get client name and client IP for logging.
+		var clientName string = "blah"
+		go serviceThread(conn, clientName)
+	}
 }
 
 func main() {
 
-        flag.Parse()
-
-        var serverAddr string
-        serverAddr = *serverHost + ":" + *serverPort
-        var arg string = "something"
-
-/*
-	REMOVE: This is a test
-        attestation, err := os.ReadFile("attestation.bin")
-        if err != nil {
-                fmt.Printf("Failed to read attestation file: %s\n", err.Error())
-        }
-
-        var what_to_say []byte
-        what_to_say = make([]byte, 256)
-        for i := 0; i < 256; i++ {
-                what_to_say[i] = byte(i)
-        }
-        outMeasurement, err := gramineverify.GramineVerify(what_to_say, attestation)
-        if err != nil {
-                fmt.Printf("GramineVerify failed: %s\n", err.Error())
-        }
-        fmt.Printf("Measurement length: %d\n", len(outMeasurement));
- */
-
-        // later this may turn into a TLS connection, we'll see
-        server(serverAddr, arg)
-        fmt.Printf("server: done\n")
+	flag.Parse()
+
+	var serverAddr string
+	serverAddr = *serverHost + ":" + *serverPort
+	var arg string = "something"
+
+	/*
+	   	REMOVE: This is a test
+	           attestation, err := os.ReadFile("attestation.bin")
+	           if err != nil {
+	                   fmt.Printf("Failed to read attestation file: %s\n", err.Error())
+	           }
+
+	           var what_to_say []byte
+	           what_to_say = make([]byte, 256)
+	           for i := 0; i < 256; i++ {
+	                   what_to_say[i] = byte(i)
+	           }
+	           outMeasurement, err := gramineverify.GramineVerify(what_to_say, attestation)
+	           if err != nil {
+	                   fmt.Printf("GramineVerify failed: %s\n", err.Error())
+	           }
+	           fmt.Printf("Measurement length: %d\n", len(outMeasurement));
+	*/
```

Comment:
Interesting ... gofmt indents-in this block commented dead code. At least that makes such code chunks stand-out ... which is sort-of ok.